nexus/src/app/background/tasks/fm_sitrep_load.rs

```diff
@@ -0,0 +1,179 @@
+    pub fn watcher(&self) -> watch::Receiver<CurrentSitrep> {
+        self.tx.subscribe()
+    }
+
+    async fn load_if_needed(&self, opctx: &OpContext) -> Status {
+        // Set up a logger for this activation that includes metadata about
+        // the current sitrep.
+        let (old, log) = match &*self.tx.borrow() {
+            None => (None, opctx.log.clone()),
+            Some(old) => {
+                let (ref old_version, _) = **old;
+                let log = opctx.log.new(slog::o!(
+                    // since this is a TypedUuid, use `Debug` to avoid
+                    // including ()
+                    "original_id" => format!("{:?}", old_version.id),
+                    "original_made_current" => old_version.time_made_current.to_string(),
+                    "original_version" => old_version.version,
+                ));
+                (Some(old_version.clone()), log)
+            }
+        };
+
+        // Get the ID of the current sitrep.
+        let time_loaded = Utc::now();
+        let current_version: SitrepVersion = match self
+            .datastore
+            .fm_current_sitrep_version(opctx)
+            .await
+        {
+            Ok(Some(version)) => version,
+            Ok(None) => match old {
+                Some(SitrepVersion { version, id, .. }) => {
+                    // We should never go from "some sitrep" to "no sitrep";
+                    // pruning should always keep a small number of old sitreps
+                    // around until we have new ones to replace them.
+                    //
+                    // In this case we won't replace our channel contents with
+                    // `None`; we'll keep around whatever old collection we had
+                    // loaded.
+                    warn!(log, "previously had a sitrep, but now none exist");
+                    return Status::Error(format!(
+                        "previously loaded sitrep {id:?} (v{version}), \
+                         but now no sitreps exist",
+                    ));
+                }
+                None => {
+                    // Had no sitrep; still have no sitrep.
+                    return Status::NoSitrep;
+                }
+            },
+            Err(err) => {
+                let err = InlineErrorChain::new(&err);
+                warn!(
+                    log,
+                    "failed to read current sitrep version";
+                    &err
+                );
+                return Status::Error(format!(
+                    "failed to read current sitrep version: {err}"
+                ));
+            }
+        };
+
+        // Have we already loaded this sitrep?
+        match old {
+            Some(version) if version.id == current_version.id => {
+                debug!(log, "current sitrep has not changed");
+                return Status::Loaded { version, time_loaded };
+            }
+            Some(SitrepVersion { version, id, .. })
```

Comment:
Arguably, the version also "shouldn't go down", right?

Idk if it's worth having special cases for these - definitely we want to detect "has the UUID changed or not", but idk if it's worth it to include match cases for invariants that should really be impossible to trigger.

Reply:
I added a check for this in dfffd6913dc38b87617b6f125524daeb689ab383